uiServices/src/testCaseServices.go

```diff
@@ -113,4 +126,66 @@ func putTestCase(rw http.ResponseWriter, req *http.Request) {
 	}
 
 	rw.WriteHeader(http.StatusNoContent)
+}
+
+func getAllTestCases(rw http.ResponseWriter, req *http.Request) {
+
+	testCasePathDir := getTestCaseHeader(req)
+	if len(testCasePathDir) <= 1 {
+		logrus.Error("No file directory entered")
+		rw.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	files, err := ioutil.ReadDir(testCasePathDir)
+	if err != nil {
+		logrus.Error(err)
+	}
+
+	testCases := make([]Case, 0)
+
+	for _, file := range files {
+		if filepath.Ext(testCasePathDir +file.Name()) == ".xml" {
+
+			testCase := new(testStrategies.TestDefinition)
+
+			filename := file.Name()
+
+			file, err := os.Open(fmt.Sprintf("%s%s", testCasePathDir, filename))
+			if err != nil {
+				logrus.Error("Cannot Open File: "+ filename)
```

Comment:
If the any of these conditions are true I would not expect the flow to continue, I would expect a log message and setResponse and error. If you can't read or unmarshall it is unrecoverable. 

```
	logrus.Error("Cannot Unmarshall")
		rw.WriteHeader(http.StatusBadRequest)
		return
```

Reply:
What happens if one of the files is being used by another user and cannot be opened? But if you need the content of all files in the directory then returning an error here makes sense.

Reply:
Our tech lead asked me to do it this way. She is away at the minute but I will ask her about this again when she returns next week. Thanks for reviewing!

Reply:
@jenny-flynn